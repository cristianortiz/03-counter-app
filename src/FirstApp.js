import React from "react";
import PropTypes from "prop-types";

const FirstApp = ({ hi2, default_prop }) => {
  const hi = { firstname: "jon", age: 34 };

  //console.log(hi2);

  return (
    <>
      {/*  <pre>{JSON.stringify(hi, null, 3)}</pre> */}
      <h3>from first app component {hi.firstname}</h3>
      <p>{hi2}</p>
      <p>{default_prop}</p>
    </>
  );
};
FirstApp.propTypes = {
  hi2: PropTypes.string.isRequired,
};

FirstApp.defaultProps = {
  default_prop: "I am by default",
};
export default FirstApp;
