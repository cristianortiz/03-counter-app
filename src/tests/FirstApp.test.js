import { shallow } from "enzyme";
import FirstApp from "../FirstApp";

describe("Test in <FirstApp component>", () => {
  /* test('should show "this is a prop value to test', () => {
    const hi2 = "this is a prop value to test";
    const { getByText } = render(<FirstApp hi2={hi2} />);
    //asses if the textt of hi2 is in the renderized component
    expect(getByText(hi2)).toBeInTheDocument(); 
  });*/

  //now using enzyme
  test("should show <FirstApp/> correctly", () => {
    const hi2 = "hello iam a greet";
    const wrapper = shallow(<FirstApp hi2={hi2} />);

    expect(wrapper).toMatchSnapshot();
  });
});
