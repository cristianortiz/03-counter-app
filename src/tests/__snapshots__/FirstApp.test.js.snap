// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in <FirstApp component> should show <FirstApp/> correctly 1`] = `
<Fragment>
  <h3>
    from first app component 
    jon
  </h3>
  <p>
    hello iam a greet
  </p>
  <p>
    I am by default
  </p>
</Fragment>
`;
